Blogpost about sprint 152 (#432)
openSUSE/obs-landing: _posts/development/2023-10-30-notifications-failed-workflow-runs.md

---
layout: post
title: Notifications about failed SCM/CI Workflows and more
category: development
---

> A lot of stuff happens behind the scenes whenever OBS runs SCM/CI workflow. Sometimes, it gets hard to keep track of things like which workflow failed or completed successfully. So, this time we tried to solve this problem along with other improvements in SCM/CI integration.

Author: I've addressed this. Please check

> A lot of things happen behind the scenes whenever OBS runs an SCM/CI workflow. Sometimes, you might want to know which workflow failed or completed successfully and the only way to do this is visiting the workflow runs UI. So, this time we tried to solve this problem along with other improvements in SCM/CI integration.

Reviewer: ```suggestion
A lot of things happen behind the scenes whenever OBS runs an SCM/CI workflow. Sometimes, you might want to know which workflow failed or completed successfully and the only way to do this is by visiting the workflow runs UI. This time we tried to improve this and other details in the SCM/CI integration.
```

{% include partials/_series-of-posts-about-scm-integration.md %}

> ## Notifications about failed SCM/CI workflows

Reviewer: Titleized:

```suggestion
## Notifications About Failed SCM/CI Workflows
```

> Whenever SCM/CI workflow executes you get a new entry of WorkflowRuns that gives all the details about status, event type, event source etc. But now OBS has the option to send notification if the WorkflowRun fails.

Reviewer: ```suggestion
Whenever an SCM/CI workflow is executed, you get a new entry of workflow run that gives all the details about status, event type, event source etc. But now OBS also gives you the possibility of receiving a notification as soon as one workflow run fails.

Give it a try! Subscribe to the new type of (notifications)[https://build.opensuse.org/my/subscriptions] so you don't have to manually check what's going on with your workflows. OBS will let you know as soon as something wrong happens.
```

<figure>

>   <img src="/images/posts/workflow-run-notification-1.png" alt="Screenshot of workflow run notification activation" />

Reviewer: This is not the correct image to display here. Email should be enabled as well, we introduced it in a later PR.

Author: The image has been updated

</figure>

<figure>
  <img src="/images/posts/workflow-run-notification-2.png" alt="Screenshot of failed workflow run" />
</figure>

> ## Workflow configuration file and SetFlag step artifacts in workflowRun

Reviewer: ```suggestion
## Workflow Configuration File and SetFlag Step Artifacts in Workflow Run
```

> Previously, WorkflowRun only showed the path of workflow configuration file and now you can check the whole content of your workflow configuration file in WorkflowRun.

Reviewer: ```suggestion
Previously, the workflow run details only included the URL or path to the workflow configuration file. Now it also includes the file content so you can always check what was the configuration at the time the workflow ran.
```

<figure>
  <img src="/images/posts/workflow-config.png" alt="Screenshot of workflow configuration file" />
</figure>

> In addition to the configuration file WorkflowRun now also stores the artifacts of SetFlag step.

Reviewer: ```suggestion
In addition to the configuration file the workflow run now also stores the artifacts of the SetFlag step.
```
